src/Symfony/Component/Translation/Command/LintCommand.php

```diff
@@ -0,0 +1,253 @@
+        $this->isReadableProvider = $isReadableProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('lint:translation')
+            ->setDescription('Lints a file and outputs encountered errors')
+            ->addArgument('filename', null, 'A file or a directory or STDIN')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The output format', 'txt')
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> command lints a XLIFF file and outputs to STDOUT
+the first encountered syntax error.
+
+You can validates XLIFF contents passed from STDIN:
+
+  <info>cat filename | php %command.full_name%</info>
+
+You can also validate the syntax of a file:
+
+  <info>php %command.full_name% filename</info>
+
+Or of a whole directory:
+
+  <info>php %command.full_name% dirname</info>
+  <info>php %command.full_name% dirname --format=json</info>
+
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $filename = $input->getArgument('filename');
+        $this->format = $input->getOption('format');
+        $this->displayCorrectFiles = $output->isVerbose();
+
+        if (!$filename) {
+            if (!$stdin = $this->getStdin()) {
+                throw new \RuntimeException('Please provide a filename or pipe file content to STDIN.');
+            }
+
+            return $this->display($io, array($this->validate($stdin)));
+        }
+
+        if (!$this->isReadable($filename)) {
+            throw new \RuntimeException(sprintf('File or directory "%s" is not readable.', $filename));
+        }
+
+        $filesInfo = array();
+        foreach ($this->getFiles($filename) as $file) {
+            $filesInfo[] = $this->validate(file_get_contents($file), $file);
+        }
+
+        return $this->display($io, $filesInfo);
+    }
+
+    private function validate($content, $file = null)
+    {
+        // Avoid: Warning DOMDocument::loadXML(): Empty string supplied as input
+        if ('' === trim($content)) {
+            return array('file' => $file, 'valid' => true);
+        }
+
+        libxml_use_internal_errors(true);
+
+        $document = new \DOMDocument();
+        $document->loadXML($content);
+        if ($document->schemaValidate(__DIR__.'/../Resources/schemas/xliff-core-1.2-strict.xsd')) {
+            return array('file' => $file, 'valid' => true);
+        }
+
+        $errorMessages = array();
+        foreach (libxml_get_errors() as $error) {
+            // general document errors have a '-1' line number
+            if (-1 === $error->line) {
+                $errorMessages[] = $error->message;
+            } else {
+                $errorMessages[] = sprintf('Line %d, Column %d: %s', $error->line, $error->column, trim($error->message));
```

Comment:
in JSON, it may be better to have the line and column as separate properties: ``{ line: 10, column: 5, message: "unexpected char"}``.
It would make the output more useful by allowing to known the location without parsing the error message.

you could do it by building such an array here, and doing the sprintf only in ``displayTxt``. In this case, I would keep the same structure for general errors, but with ``null`` for the line and column.